config/routes.rb

```diff
@@ -78,6 +78,15 @@
       end
     end
 
+    resources :messages, :path => "user/messages", :constraints => { :id => /\d+/ }, :only => [:create, :show, :destroy], :controller => "messages", :as => :api_messages do
```

Comment:
Do we want to nest messages under user? The IDs are global so there's no need for it unless we think there's some reason it makes more sense that way.

Reply:
In general I would not recommend any nesting. I would only add the nesting if either:
* There's a need for the id of the parent object (e.g. /node/:id/relations), or
* There's a need to disambiguate between two same-named controllers (e.g. the classic `/admin/users/`)

However, in this particular case, I support @milan-cvetkovic , since it actually matches our existing API. As far as I can tell, almost every existing call where the responses are specific to the current user (and don't refer to public objects) are nested under `/user`:
* /api/0.6/user/details
* /api/0.6/user/gpx_files
* /api/0.6/user/preferences

However some are not:
* /api/0.6/permissions

(There are other calls where it's a user-specific action on an otherwise public object, like subscribing to a changeset, but I treat these as a different situation than the above).

So I think we should add the nesting for messages too, purely for consistency, even though I don't like it!

Reply:
If @gravitystorm thinks it is, on balance, the right choice then I think we can leave this as it is.